app/ui-react/syndesis/src/modules/integrations/components/editor/EditorSteps.tsx

```diff
@@ -0,0 +1,99 @@
+import { getMetadataValue } from '@syndesis/api';
+import * as H from '@syndesis/history';
+import { ConnectionOverview } from '@syndesis/models';
+import {
+  ConnectionCard,
+  ConnectionsGrid,
+  ConnectionsGridCell,
+  ConnectionSkeleton,
+} from '@syndesis/ui';
+import { WithLoader } from '@syndesis/utils';
+import * as React from 'react';
+import { Translation } from 'react-i18next';
+import { ApiError, EntityIcon } from '../../../../shared';
+import { IUIStep } from './interfaces';
+
+export interface IEditorStepsProps {
+  error: boolean;
+  loading: boolean;
+  steps: IUIStep[];
+
+  getConnectionHref(connection: ConnectionOverview): H.LocationDescriptor;
```

Comment:
we are not dealing with just connections, so better not use `connection` when naming things. Also, the object that's passed back should really be a `IUIStep`, shouldn't it?